django/contrib/messages/middleware.py

```diff
@@ -24,3 +24,9 @@ def process_response(self, request, response):
             if unstored_messages and settings.DEBUG:
                 raise ValueError('Not all temporary messages could be stored.')
         return response
+
+    def __repr__(self):
+        return '<%(cls)s MESSAGE_STORAGE="%(storage)s">' % {
+            'cls': self.__class__.__name__,
+            'storage': settings.MESSAGE_STORAGE,
```

Comment:
The storage here is really only relevant if `request._messages` is set, right? If it's not, then perhaps you can have a default "unknown" string or something. I agree with Tim that referencing settings in middleware repr methods is probably not the right thing to be doing.